src/api/currencyConfig.js

```diff
@@ -0,0 +1,40 @@
+// @flow
+
+import {DEFAULT_SUFFIX} from "../ledger/grain";
+import * as C from "../util/combo";
+
+/**
+ * Shape of concurrencyDetails.json on disk
+ */
+type SerializedCurrencyDetails = {|
+  +currencyName?: string,
+  +currencySuffix?: string,
+|};
+
+/**
+ * Shape of currencyDetails json in memory after parsing
+ */
+export type CurrencyDetails = {|
+  +name: string,
+  +suffix: string,
+|};
+
+export const DEFAULT_NAME = "Grain";
+
+/**
+ * Utilized by combo.fmap to enforce default currency values
+ * when parsing. This engenders a "canonical default" since there
+ * will be no need to add default fallbacks when handling currency
+ * detail values after parsing the serialized file.
+ */
+function upgrade(c: SerializedCurrencyDetails): CurrencyDetails {
+  return {
+    name: c.currencyName || DEFAULT_NAME,
+    suffix: c.currencySuffix || DEFAULT_SUFFIX,
```

Comment:
nit: I generally prefer `c.currencyName == null ? DEFAULT_NAME : c.currencyName` or (alternatively) `NullUtil.orElse(c.currencyName, DEFAULT_NAME)`. 

That way we treat all input strings consistently, rather than having an implicit special case where if the user specifies the empty string we refuse to respect the user's choice and treat it as unset instead. It comes from the general principle in our codebase of trying to treat the empty element of a type consistently wherever possible.

(change not required).